### PR TITLE
Debug Print Function 

### PR DESCRIPTION
**Pull request by shyammukund**
Help debug issue where the deque does not print the full contents after you append right first then try and append left. The count shows that all the elements are present in the deque which is why I believe that the issue lies within the print function.

### CODE REVIEW

**Comment by shyammukund** [comment]
@dnukum

**Comment by shyammukund** [comment]
@dnukum

File: deque/main_test.cpp [hunk 1]
```diff
@@ -211,12 +211,13 @@ TEST_CASE("DequeList<T>::append_left_bug()", "[DequeList]") {
...
   for (auto& e : v1) {
     d1.append_left(e);
   }
   vector<int> resd1 = d1.print();
-  REQUIRE(resd1 == v1);
+  vector<int> exp = {10, 20, 30, 40, 50, 60, 70, 80, 90};
+  REQUIRE(resd1 == exp);
   REQUIRE(d1.get_count() == (int)v1.size());
 
 }
```
Comment (dnukum) [inline deque/main_test.cpp:240]: TEST_CASE("DequeList<T>::operator=", "[DequeList]") {
    DequeList<int> d1(4);
    vector<int> v1{10, 20, 30, 40};
    for (auto& e : v1) {
        d1.append_left(e);
    } 

    DequeList<int> d2(4);
    d2 = d1;
    auto r1 = d1.print();
    auto r2 = d2.print();
    REQUIRE (r1 == r2);
}

the test is failing.

File: deque/dequeList.hpp [hunk 2]
```diff
@@ -301,44 +259,6 @@ class DequeList {
...
-           else {
-               back_index++;
-           }
-           // Find the current end block where we need to add the element
-           map[end_block][back_index] = element;
-       }
-       count++;
-   } */
-
   T pop_left() {
```
Comment (dnukum) [inline deque/dequeList.hpp:262]: Incorrect Indexing in pop_left and pop_right:
 The methods pop_left and pop_right incorrectly update start_block and end_block   using modulo operations that assume the blocks wrap around within the map array. This is not a correct way to manage these indices, as it doesn't account for the actual logical sequence of blocks.  Instead, decrementing and incrementing start_block and end_block directly, while managing when to reset or wrap the indices, would be more appropriate.

**Review by dnukum — COMMENTED** [review]
Check the at function. Boundary conditions not checked

File: deque/dequeList.hpp (hunk 2 above)
Comment (dnukum) [reply deque/dequeList.hpp:262]: Also at function is not checking boundary conditions.

File: deque/dequeList.hpp [hunk 3]
```diff
@@ -5,360 +5,383 @@
...
+
+  // --------------------------------------------------------------------
+  //
+  // Implementation of public functions
+  //
+  // --------------------------------------------------------------------
+
+ public:
+  // Constructor
+  // TO DO: change map_capacity to be min capacity that the deque can hold at
```
Comment (dnukum) [inline deque/dequeList.hpp:76]: Remove all the TODO

File: deque/dequeList.hpp [hunk 4]
```diff
@@ -5,360 +5,383 @@
...
-
-    // TO DO: Add additional helper functions as needed
-
-// --------------------------------------------------------------------
-//
-// Implementation of public functions
-//
-// --------------------------------------------------------------------
+class DequeList {
+  /* public:
```
Comment (dnukum) [inline deque/dequeList.hpp:9]: Remove dead code

File: deque/dequeList.hpp [hunk 5]
```diff
@@ -5,360 +5,383 @@
...
-    // Destructor
-    
-    ~DequeList()
-    {
-        for (int i = 0; i < map_size; i++){
-            delete[] map[i];
-        }
-        delete [] map;
+  // copy assignment operator
+  DequeList& operator=(const DequeList& other) {
```
Comment (dnukum) [inline deque/dequeList.hpp:122]: You need test for this. There is a bug here.

**Pull request merged**